src/sentry/snuba/query_subscription_consumer.py

```diff
@@ -58,42 +72,59 @@ def inner(func: TQuerySubscriptionCallable) -> TQuerySubscriptionCallable:
     return inner
 
 
-def parse_message_value(value: str) -> Dict[str, Any]:
+def parse_message_value(value: str, topic: str) -> PayloadV3:
     """
     Parses the value received via the Kafka consumer and verifies that it
     matches the expected schema.
     :param value: A json formatted string
     :return: A dict with the parsed message
     """
+    SUPPORTED_SCHEMA_VERSION = 3
+    old_version = False
+
     with metrics.timer("snuba_query_subscriber.parse_message_value.json_parse"):
-        wrapper: Dict[str, Any] = json.loads(value)
+        wrapper: SubscriptionResults = json.loads(value)
 
     with metrics.timer("snuba_query_subscriber.parse_message_value.json_validate_wrapper"):
         try:
-            jsonschema.validate(wrapper, SUBSCRIPTION_WRAPPER_SCHEMA)
-        except jsonschema.ValidationError:
-            metrics.incr("snuba_query_subscriber.message_wrapper_invalid")
-            raise InvalidSchemaError("Message wrapper does not match schema")
+            jsoncodec = JsonCodec(get_schema(topic)["schema"])
```

Comment:
You shouldn't compile the json codec in here on every message. It's slow and it's always the same. Just do it once outside this function and then call `.validate()` on each message.